pkg/controller/util/wildfly_mgmt.go

```diff
@@ -0,0 +1,129 @@
+package util
+
+import (
+	"bytes"
+	"crypto/md5"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+)
+
+var (
+	// MgmtOpServerStateRead is a JSON structure for reading WFLY server
+	MgmtOpServerStateRead = map[string]interface{}{
+		"address":   []string{},
+		"operation": "read-attribute",
+		"name":      "server-state",
+	}
+	// MgmtOpReload is a JSON structure for reloading WFLY server
+	MgmtOpReload = map[string]interface{}{
+		"address":   []string{},
+		"operation": "reload",
+	}
+	// MgmtOpTxnEnableRecoveryListener is a JSON structure for enabling txn recovery listener
+	MgmtOpTxnEnableRecoveryListener = map[string]interface{}{
+		"address": []string{
+			"subsystem", "transactions",
+		},
+		"operation": "write-attribute",
+		"name":      "recovery-listener",
+		"value":     "true",
+	}
+	// MgmtOpTxnProbe is a JSON structure for probing transaction log store
+	MgmtOpTxnProbe = map[string]interface{}{
+		"address": []string{
+			"subsystem", "transactions", "log-store", "log-store",
+		},
+		"operation": "probe",
+	}
+	// MgmtOpTxnRead is a JSON structure for reading transaction log store
+	MgmtOpTxnRead = map[string]interface{}{
+		"address": []string{
+			"subsystem", "transactions", "log-store", "log-store",
+		},
+		"operation":       "read-children-resources",
+		"child-type":      "transactions",
+		"recursive":       "true",
+		"include-runtime": "true",
+	}
+)
+
+// ConvertJSONToReader converts the provided JSON data (a map)
+//  to io.Reader which can be consumed by methods sending via HTTP
+func convertJSONToReader(jsonData map[string]interface{}) (io.Reader, error) {
+	jsonStreamBytes, err := json.Marshal(jsonData)
+	if err != nil {
+		return nil, fmt.Errorf("Fail to marshal JSON message %v", jsonData)
+	}
+	return bytes.NewBuffer(jsonStreamBytes), nil
+}
+
+// GenerateWflyMgmtHashedPassword generate the password in form to be saved under mgmt-users.properties
+func GenerateWflyMgmtHashedPassword(user, password string) string {
+	data := []byte(user + ":ManagementRealm:" + password)
+	return fmt.Sprintf("%x", md5.Sum(data))
+}
+
+// IsMgmtOutcomeSuccesful verifies if the HTTP based management operation was succcesfull
+//   it expects on arguments to get the response from the HTTP base operation (res)
+//   and the body returned from the HTTP operation that is represented as a JSON
+func IsMgmtOutcomeSuccesful(jsonBody map[string]interface{}) bool {
+	return jsonBody["outcome"] == "success"
+}
+
+// ExecuteMgmtOp executes WildFly managemnt operation represented as a JSON
+//  the execution runs over the HTTPDigest struct
+//  returns the JSON as the return value from the HTTP
+func ExecuteMgmtOp(httpDigest *HTTPDigest, mgmtOpJSON map[string]interface{}) (map[string]interface{}, error) {
```

Comment:
the `CLI` scripts are not needed in this version. My intention was to start using the WFLY HTTP management interface as I think it's a cleaner way to access the management operation. The `ExecRemote` is then more like passing a concatenated strings to some bash operation. Second thing why I think it could be better is that it does not depend on the underlying operating system. If there would be e.g. system without shell script or Windows OS then access via wfly management is independent on that.

This all comes from my strong feeling that operator should be independent on the s2i images. It should cooperate with them smoothly, it should be tested with them but it should not be required to use them for operator to work.

But still there is need the `ExecRemote` call. And the reason is that there is a need to check the content of the data directory of the wfly transaction client. There is no management operation to get information if all remote transactions from the library were processed. Thus in this version of the operator the remote call is needed.

Reply:
I don't understand your concern with the scripts. They are running in the container which is Linux, the script will always reliably be executed. 

S2I *is* a requirement from the operator as most of its features will actually be implemented by S2I.

Reply:
Ok. My concern is that the operator should be the most independent as possible. My vision would be possibility to run operator with s2i (that brings some more features) but still being possible to run it *without* s2i (that provides basic features - like scaling, ensures data consistency...). 

As you consider the s2i necessary (just I'm not sure what are those features where s2i is the requirements in current operator) I'm going to change the code to run only the *ssh remote calls* and won't be using the http management calls.